src/Trace/Span.php

```diff
@@ -0,0 +1,507 @@
+ * and form a trace tree. Often, a trace contains a root span that describes
+ * the end-to-end latency of an operation and, optionally, one or more subspans
+ * for its suboperations. Spans do not need to be contiguous. There may be
+ * gaps between spans in a trace.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Trace\TraceClient;
+ *
+ * $traceClient = new TraceClient();
+ * $trace = $traceClient->trace();
+ * $span = $trace->span([
+ *     'name' => 'span name',
+ *     'attributes' => ['foo' => 'bar'],
+ *     'stackTrace' => debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS)
+ * ]);
+ * $span->setStartTime();
+ * $span->setEndTime();
+ * ```
+ * @codingStandardsIgnoreStart
+ * @see https://cloud.google.com/trace/docs/reference/v2/rest/v2/projects.traces/batchWrite#Span Span model documentation
+ * @codingStandardsIgnoreEnd
+ */
+class Span implements \JsonSerializable
+{
+    use AttributeTrait;
+    use TimestampTrait;
+
+    /**
+     * @var string [TRACE_ID] is a unique identifier for a trace within a
+     *      project; it is a 32-character hexadecimal encoding of a 16-byte
+     *      array.
+     */
+    private $traceId;
+
+    /**
+     * @var string The [SPAN_ID] portion of the span's resource name.
+     */
+    private $spanId;
+
+    /**
+     * @var string The [SPAN_ID] of this span's parent span. If this is a root
+     *      span, then this field must be empty.
+     */
+    private $parentSpanId;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string The start time of the span. On the client side,
+     *      this is the time kept by the local machine where the span execution
+     *      starts. On the server side, this is the time when the server's
+     *      application handler starts running.
+     */
+    private $startTime;
+
+    /**
+     * @var string The end time of the span. On the client side,
+     *      this is the time kept by the local machine where the span execution
+     *      ends. On the server side, this is the time when the server's
+     *      application handler stops running.
+     */
+    private $endTime;
+
+    /**
+     * @var array Stack trace captured at the start of the span.
+     */
+    private $stackTrace;
+
+    /**
+     * @var TimeEvent[] A set of time events. You can have up to 32 annotations
+     *      and 128 message events per span.
+     */
+    private $timeEvents;
+
+    /**
+     * @var Link[] Links associated with the span. You can have up to 128 links
+     *      per Span.
+     */
+    private $links;
+
+    /**
+     * @var Status An optional final status for this span.
+     */
+    private $status;
+
+    /**
+     * @var bool A highly recommended but not required flag that identifies when
+     *      a trace crosses a process boundary. True when the parent_span
+     *      belongs to the same process as the current span.
+     */
+    private $sameProcessAsParentSpan;
```

Comment:
This property isn't used in this class.

Reply:
Added it back to the constructor and the jsonSerialize